test/backend/test_node.py

```diff
@@ -1056,11 +1057,23 @@ def test_hardmax(self):
     for axis in range(-len(shape), len(shape)):
       node_def = helper.make_node("Hardmax", ["X"], ["Y"], axis=axis)
       output = run_node(node_def, [x])
-      shape_in_2d = (np.prod(shape[0:axis]).astype(int),
-                     np.prod(shape[axis:len(shape)]))
-      x_in_2d = np.reshape(x, shape_in_2d)
-      y = np.eye(x_in_2d.shape[1], dtype=x.dtype)[np.argmax(x_in_2d, axis=1)]
-      np.testing.assert_almost_equal(output["Y"], np.reshape(y, shape))
+
+      axis = axis if axis >= 0 else len(np.shape(x)) + axis
+      if axis == len(np.shape(x)) - 1:
+          np.testing.assert_almost_equal(output["Y"], tfa.seq2seq.hardmax(x))
+      else:
+          if not legacy_opset_pre_ver(13):
+            perm1 = tf.range(0, axis)
+            perm2 = tf.range(axis + 1, len(tf.shape(x)) - 1)
+            perm = tf.concat([perm1, [len(tf.shape(x)) - 1], perm2, [axis]], -1)
+            x_transposed = tf.transpose(x, perm)
+            np.testing.assert_almost_equal(output["Y"], tf.transpose(tfa.seq2seq.hardmax(x_transposed), perm))
```

Comment:
@shahirdaya how about using hardmax defined in onnx.backend.test.case.node.hardmax to calculate the expected value here, then it will be using different way to calculate the expected value and compare the value return from the hardmax handler.

Reply:
Thanks @winnietsang. Done.